packages/core/utils/src/dml/helpers/mikro-orm/apply-indexes.ts

```diff
@@ -16,6 +16,7 @@ export function applyIndexes(
 ) {
   field.indexes.forEach((index) => {
     const providerEntityIdIndexStatement = createPsqlIndexStatementHelper({
+      name: index.name,
```

Comment:
@olivermrbl Because of this, the migration was dropping an existing index and was re-creating it with a different name